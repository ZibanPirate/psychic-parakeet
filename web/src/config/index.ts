import { Environment } from "@parakeet/api/src/config/dto";

/**
 * Get the current environment
 */
export const getEnv: () => Environment = () => {
  switch (location.hostname) {
    case "localhost":
      return "development";
    default:
      return "production";
  }
};

const frontendPort = 8080;
const apiPort = 7070;
const environments: Environment[] = ["development", "production"];
const e = environments.indexOf(getEnv());

/**
 * Configuration object, containing global configuration and variable based on current environment
 */
export const config = {
  api: {
    url: [`http://localhost:${apiPort}`, "https://api_parakeet.zakiii.com"][e],
  },
  web: {
    port: frontendPort,
    url: [`http://localhost:${frontendPort}`, "https://parakeet.zakiii.com"][e],
  },
};
