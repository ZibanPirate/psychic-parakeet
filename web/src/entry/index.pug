doctype html
head(lang=lang)
  meta(charset="utf-8")
  meta(http-equiv="X-UA-Compatible", content="IE=edge")
  meta(name="viewport", content="width=device-width, initial-scale=1")

  title= title
  link(rel="icon", href=require('src/assets/ico/logo-square.ico').default)
  meta(name="description", content=description)
  meta(name="keywords", content=keywords)
  meta(property="og:image", content=ogImage)
  meta(name="theme-color", content=themeColor)

  style.
    body {
      margin: 0;
      padding: 0;
    }
    #splash-screen {
      position: absolute;
      width: 100vw;
      height: 100vh;
      background: #282c34;
    }
    #splash-screen::after {
      content: '';
      position: absolute;
      left: 50%;
      top: 50%;
      transform: translate(-50%, -50%);

      width: 80%;
      height: 80%;

      background-image: url("#{require('!!base64-inline-loader!../assets/svg/logo-wide.svg')}");
      background-position: center;
      background-size: contain;
      background-repeat: no-repeat;
      max-width: 600px;
    }
    body > #splash-screen {
      z-index: 1000;
    }
    * {
      font-family: 'Tajawal', -apple-system, BlinkMacSystemFont, 'Segoe UI', 'PingFang SC', 'Hiragino Sans GB', 'Microsoft YaHei', 'Helvetica Neue', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
    }
  
body
  #splash-screen
  #app-container
    #splash-screen
