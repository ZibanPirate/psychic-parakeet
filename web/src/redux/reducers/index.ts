import { combineReducers } from "redux";
import { landingPage } from "./landing-page";
import { settings } from "./settings";

export const mainReducer = combineReducers({
  settings,
  landingPage,
});
