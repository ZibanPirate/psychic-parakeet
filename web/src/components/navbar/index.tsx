import { Dispatch, StateInterface } from "src/redux";
import { createStyles, makeStyles } from "@material-ui/core/styles";
import { useDispatch, useSelector } from "react-redux";
import AppBar from "@material-ui/core/AppBar";
import { FC } from "react";
import { SettingsState } from "src/redux/reducers/settings";
import Switch from "@material-ui/core/Switch";
import Toolbar from "@material-ui/core/Toolbar";

const useStyles = makeStyles(() =>
  createStyles({
    darkModeText: {
      flexGrow: 1,
    },
  }),
);

export const Navbar: FC = () => {
  const state = useSelector<StateInterface, StateInterface>((state) => state);
  const dispatch = useDispatch<Dispatch<SettingsState>>();
  const classes = useStyles();

  return (
    <AppBar position="static">
      <Toolbar>
        <Switch
          checked={state.settings.darkMode ? true : false}
          onChange={(input) => {
            dispatch({
              type: "UPDATE_SETTINGS",
              payload: { darkMode: input.target.checked },
            });
          }}
          name="darkMode"
          inputProps={
            // there is a bug in https://github.com/mui-org/material-ui/tree/next/packages/material-ui/src/Switch
            // i will probably open an issue for it, then fix it on an MR, the interface should be InputBaseComponentProps
            // instead of React.InputHTMLAttributes<HTMLInputElement>, so that it can allow arbitrary attributes such as "data-testid"
            {
              "data-testid": "dark-mode-switch",
            } as React.InputHTMLAttributes<HTMLInputElement>
          }
        />
        <div className={classes.darkModeText}>
          {state.settings.darkMode ? "ðŸŒ™" : "ðŸŒž"}
        </div>
      </Toolbar>
    </AppBar>
  );
};
