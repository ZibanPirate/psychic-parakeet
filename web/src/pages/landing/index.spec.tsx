import {
  cleanup,
  fireEvent,
  render,
  screen,
  waitFor,
} from "@testing-library/react";
import { LandingPage } from ".";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { Theme } from "src/components/theme";
import { createMainStore } from "src/redux";

describe("Landing Page", () => {
  afterEach(() => cleanup());
  test("Render Landing page", async () => {
    const mainStore = createMainStore();

    const { container } = render(
      <Provider store={mainStore}>
        <Router>
          <Theme>
            <LandingPage />
          </Theme>
        </Router>
      </Provider>,
    );
    await waitFor(() => screen.getByText("Search goes here"));

    expect(container).toMatchSnapshot();
  });

  test("Render Landing page after switching darkMode on", async () => {
    const mainStore = createMainStore();

    const { container } = render(
      <Provider store={mainStore}>
        <Router>
          <LandingPage />
        </Router>
      </Provider>,
    );

    await waitFor(() => screen.getByText("ðŸŒ™"));

    const darkModeModeSwitch = screen.getByTestId("dark-mode-switch");

    fireEvent.click(darkModeModeSwitch);

    await waitFor(() => screen.getByText("ðŸŒž"));

    expect(container).toMatchSnapshot();
  });
});
