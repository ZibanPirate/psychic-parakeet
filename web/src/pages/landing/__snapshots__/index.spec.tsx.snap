// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Render Landing page 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <span
        class="MuiSwitch-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-4 Mui-checked"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-6 MuiSwitch-input"
              data-testid="dark-mode-switch"
              name="darkMode"
              type="checkbox"
              value=""
            />
            <span
              class="MuiSwitch-thumb"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track"
        />
      </span>
      <div
        class="makeStyles-darkModeText-2"
      >
        ðŸŒ™
      </div>
    </div>
  </header>
  <div
    class="makeStyles-searchContainer-1"
  >
    <div>
      Search goes here
    </div>
    <div>
      Found 
      0
    </div>
  </div>
</div>
`;

exports[`Landing Page Render Landing page after switching darkMode on 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <span
        class="MuiSwitch-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-9 MuiSwitch-switchBase MuiSwitch-colorSecondary"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-12 MuiSwitch-input"
              data-testid="dark-mode-switch"
              name="darkMode"
              type="checkbox"
              value=""
            />
            <span
              class="MuiSwitch-thumb"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track"
        />
      </span>
      <div
        class="makeStyles-darkModeText-8"
      >
        ðŸŒž
      </div>
    </div>
  </header>
  <div
    class="makeStyles-searchContainer-7"
  >
    <div>
      Search goes here
    </div>
    <div>
      Found 
      0
    </div>
  </div>
</div>
`;
